pypiserver/_app.py

```diff
@@ -104,7 +91,7 @@ def root():
     fp = request.custom_fullpath
 
     try:
-        numpkgs = len(list(packages()))
+        numpkgs = len(list(core.get_all_packages()))
```

Comment:
It would be nice if we added a "package_count" or similar method to the backend so that, for example, we wouldn't actually have to parse all the files into `PkgFile` instances in order to get their count, and could instead just nab all files whose names match a regex or something. It would also be useful with imagined remote backends, where pulling all the files might be a problem

Reply:
a `package_count` method sounds good, but I'm not sure about adding "query" or filter functionality. This would require additional complexity from each backend (including said imagined ones) without adding apparent value. What would be a use case for having such filtering functionality in a `package_count` method?

Reply:
Sorry, I just meant a count without any custom filtering, but I expressed that poorly. Where I was talking about a regex, I essentially just meant "count every file that looks like a package based on the filename"